bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/Activator.java

```diff
@@ -11,6 +11,7 @@
 import org.eclipse.smarthome.core.common.osgi.ServiceBinder.Bind;
 import org.eclipse.smarthome.core.common.osgi.ServiceBinder.Unbind;
 import org.eclipse.smarthome.core.thing.type.ChannelTypeRegistry;
+import org.eclipse.smarthome.core.thing.type.DynamicStateDescriptionRegistry;
```

Comment:
Do you really need the code in this `Activator` class?

Instead the `ChannelStateDescriptionProvider` could get a service reference
```
@Reference
protected void setDynamicStateDescriptionRegistry(DynamicStateDescriptionRegistry dynamicStateDescriptionRegistry) {
        this.dynamicStateDescriptionRegistry = dynamicStateDescriptionRegistry;
}
```